code/Control/UserDefinedFormController.php

```diff
@@ -255,8 +256,13 @@ public function process($data, $form)
 
                         // create the file from post data
                         $upload = Upload::create();
-                        $file = File::create();
+
+                        $extension = File::get_file_extension($_FILES[$field->Name]['name']);
+                        $fileClass = File::get_class_for_file_extension($extension);
```

Comment:
This method was copied from https://github.com/silverstripe/silverstripe-asset-admin/blob/1/code/Controller/AssetAdmin.php#L302

Please flag this is you think this is a potential security issue.  I'm not personally sure if it's a problem if a non-image is saved as an image.

If so, [mimevalidator](https://github.com/silverstripe/silverstripe-mimevalidator) would be probably be the logical solution, though we'd need to release this as a new minor if we made that a dependency, and will increase the scope of this work

Reply:
Have discussed internally, adding mimevalidator seems like it may be a fairly good option given it has broader security benefits. If we do this it will probably be as a separate PR, so will hold off making any decisions in regards to this work for now.